### PR TITLE
fix(v3): Misc. fixes for `next` branch

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

- Standardized CI workflows to use `pnpm@8`
- Streamlined resource URI logic for spool nodes
- Removed old JobSpoolProvider (again) and its refs
- Fixed behavior for `getProviderForNode` function - resolves not being able to add favorites in DS or USS trees
- Added type guards for USS and Data Set nodes
- Marked `IZoweTreeNode.getUSSDocumentFilePath` as deprecated as Zowe Explorer no longer uses it

## Release Notes

Milestone: vNext

Changelog:
- Deprecated the `getUSSDocumentFilePath` function on the `IZoweTreeNode` interface as Zowe Explorer no longer uses the local file system for storing USS files. **No replacement is planned**; please access data from tree nodes using their [resource URIs](https://github.com/zowe/zowe-explorer-vscode/wiki/FileSystemProvider#operations-for-extenders) instead. [#2968](https://github.com/zowe/zowe-explorer-vscode/pull/2968)
- Fixed regression where `getProviderForNode` returned the wrong tree provider after a user performed an action on a Zowe tree node. [#2967](https://github.com/zowe/zowe-explorer-vscode/issues/2967)

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found

### CODE REVIEW

**Review by anaxceron — DISMISSED** [review]
Looks good, thanks @traeok!

**Review by JillieBeanSim — DISMISSED** [review]
LGTM! thanks @traeok

**Review by anaxceron — DISMISSED** [review]

**Review by zFernand0 — DISMISSED** [review]
sorry for causing merge conflicts 😢 
The changes do make sense to me though 😋 
LGTM! 🥳

File: packages/zowe-explorer/src/utils/SpoolUtils.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export class SpoolUtils {
      * @returns Array of spool files
      */
     public static async getSpoolFiles(node: IZoweJobTreeNode): Promise<zosjobs.IJobFile[]> {
```
Comment (t1m0thyj) [inline packages/zowe-explorer/src/utils/SpoolUtils.ts:23]: The `matchSpool` function in this file (and associated unit tests) should be updated to match the new implementation from this PR: https://github.com/zowe/zowe-explorer-vscode/pull/2943

File: packages/zowe-explorer/src/trees/job/JobTree.ts [hunk 2]
```diff
@@ -1061,13 +1061,6 @@ export class JobTree extends ZoweTreeProvider<IZoweJobTreeNode> implements Types
         const pollInterval = await this.showPollOptions(node.resourceUri);
 
         if (pollInterval === 0) {
-            Gui.showMessage(
```
Comment (t1m0thyj) [inline packages/zowe-explorer/src/trees/job/JobTree.ts:1064]: Curious why this message was removed - do we consider it to be superfluous?

File: packages/zowe-explorer-api/CHANGELOG.md [hunk 3]
```diff
@@ -34,6 +34,7 @@ All notable changes to the "zowe-explorer-api" extension will be documented in t
 - **Breaking:** Removed the redundant `zoweFileProvider` parameter from `editSession` method on the `IZoweTree` interface. [#2947](https://github.com/zowe/zowe-explorer-vscode/pull/2947)
 - Updated the `ProfilesCache.getProfiles` method to return empty list instead of undefined when there are no profiles. [#2947](https://github.com/zowe/zowe-explorer-vscode/pull/2947)
 - Added PEM certificate support as an authentication method for logging into the API ML. [#2621](https://github.com/zowe/zowe-explorer-vscode/issues/2621)
+- Deprecated the `getUSSDocumentFilePath` function on the `IZoweTreeNode` interface as Zowe Explorer no longer uses the local file system for storing USS files. **No replacement is planned**; please access data from tree nodes using their [resource URIs](https://github.com/zowe/zowe-explorer-vscode/wiki/FileSystemProvider#operations-for-extenders) instead. [#2968](https://github.com/zowe/zowe-explorer-vscode/pull/2968)
```
Comment (t1m0thyj) [inline packages/zowe-explorer-api/CHANGELOG.md:37]: Please add this to `docs/early-access/v3/Extenders.md` 🙂

**Review by t1m0thyj — CHANGES_REQUESTED** [review]
Thanks @traeok for fixing the regression with SharedTreeProviders!

File: packages/zowe-explorer/src/trees/job/JobTree.ts (hunk 2 above)
Comment (traeok) [reply packages/zowe-explorer/src/trees/job/JobTree.ts:1064]: Yep, we had some discussions in the past about the redundancy of "cancelled dialogs" as they don't really mesh well with the VS Code UX. I encountered this one while testing polling with the changes from this PR, so I figured it was best to remove it.

File: packages/zowe-explorer-api/CHANGELOG.md (hunk 3 above)
Comment (traeok) [reply packages/zowe-explorer-api/CHANGELOG.md:37]: Thanks for the reminder, addressed in c77d596 👍

File: packages/zowe-explorer/src/utils/SpoolUtils.ts (hunk 1 above)
Comment (traeok) [reply packages/zowe-explorer/src/utils/SpoolUtils.ts:24]: Addressed in eaa7380 😋

**Review by t1m0thyj — COMMENTED** [review]
LGTM, thanks @traeok!

**Review by zFernand0 — APPROVED** [review]

**Review by t1m0thyj — APPROVED** [review]

**Pull request merged**